tests/unittests/algo/test_evolution_es.py

```diff
@@ -0,0 +1,284 @@
+
+        # Winner team will be [0, 2], so [0, 2] will be remained, [1, 3] will be mutated.
+        assert org_data.shape == mutated_data.shape
+        assert (mutated_data[0] == org_data[0]).all()
+        assert (mutated_data[2] == org_data[2]).all()
+        assert (mutated_data[1] != org_data[1]).any()
+        assert (mutated_data[3] != org_data[3]).any()
+        assert (mutated_data[1] != org_data[0]).any()
+        assert (mutated_data[3] != org_data[2]).any()
+
+        # For each individual, mutation occurs in only one dimension chosen from two.
+        if mutated_data[1][0] != org_data[0][0]:
+            assert mutated_data[1][1] == org_data[0][1]
+        else:
+            assert mutated_data[1][1] != org_data[0][1]
+
+        if mutated_data[3][0] != org_data[2][0]:
+            assert mutated_data[3][1] == org_data[2][1]
+        else:
+            assert mutated_data[3][1] != org_data[2][1]
+
+    def test_duplicated_mutated_population(self, bracket, rung_4):
+        """Verify duplicated candidates can be found and processed correctly."""
+        red_team = [0, 2]
+        blue_team = [0, 2]  # no mutate occur at first.
+        population_range = 4
+        for i in range(4):
+            for j in [1, 2]:
+                bracket.eves.population[j][i] = list(rung_4["results"].values())[i][1][j]
+        points, nums_all_equal = bracket._mutate_population(red_team, blue_team,
+                                                            rung_4["results"], population_range)
+
+        # In this case, duplication will occur, and we can make it mutate one more time.
+        # The points 1 and 2 should be different, while one of nums_all_equal should be 1.
+        if points[1][1] != points[2][1]:
+            assert points[1][2] == points[2][2]
+        else:
+            assert points[1][2] != points[2][2]
+
+        assert nums_all_equal[0] == 0
+        assert nums_all_equal[1] == 0
+        assert nums_all_equal[2] == 1
+        assert nums_all_equal[3] == 0
+
+    def test_mutate_points(self, bracket, rung_3):
+        """Test that correct point is promoted."""
+        red_team = [0, 2]
+        blue_team = [0, 2]
+        population_range = 4
+        for i in range(4):
+            for j in [1, 2]:
+                bracket.eves.population[j][i] = list(rung_3["results"].values())[i][1][j]
+        points, nums_all_equal = bracket._mutate_population(red_team, blue_team,
+                                                            rung_3["results"], population_range)
+        assert points[0] == (1.0, 1.0, 1.0)
+        assert points[1] == (2, 1.0 / 2, 1.0 / 4)
+        assert (nums_all_equal == 0).all()
+
+    def test_customized_mutate_population(self, bracket, rung_3):
+        """Verify mutated candidates is generated correctly."""
+        red_team = [0, 2]
+        blue_team = [1, 3]
+        population_range = 4
+
+        mutate_attr = {}
+        function_string = mutate_attr.pop('function',
```

Comment:
What I meant was to replace this part with 
```python
algo = EvolutionES(space, mutate="orion.core.utils.tests.customized_mutate_example")
```
And then use `algo.brackets[0]._mutate_population` to test the custom function. This way we test the code within `EvolutionES` to convert the string to a function and the test won't need to be adapted to reflect changes in EvolutionES builder if we make any.

Reply:
I see. Done.